dhall-docs/README.md

```diff
@@ -97,6 +60,377 @@ dhall-docs --input . --output-link ${OTHER_LINK}
 
 For more information about the tool, check the `--help` flag.
 
+
+## Documenting your packages
+
+Documenting your package is essentially writing comments on your source code,
+although there are some format rules that these needs to obey in order to work
+properly with `dhall-docs`. These rules aid `dhall-docs` in extracting the text
+on the documentation that will be passed to our Markdown preprocessor to finally
+render your documentation in HTML.
+
+You can see examples of writing documentation on the [tests](./tasty/data/package)
+folder.
+
+### Documentation markup language
+
+The markup language of the documentation is [CommonMark], which is a strict
+Markdown flavour. `dhall-docs` uses [`mmark`], a package that parses
+[CommonMark] and transforms it to HTML. This package follows (almost)
+that specification.
+
+### Block comments
+
+Normal block comments in Dhall starts with `{-` and ends with `-}`. For example:
+
+```dhall
+{- foo
+     bar
+       baz
+-}
+```
+
+A `dhall-docs` block comment needs to start with `"{-|\n"` i.e. a `{-|` and a
+newline. This means that the actual documentation _must_ start on the following line.
```

Comment:
In analogy to multi-line strings, we should also support `\r\n` line-endings:

https://github.com/dhall-lang/dhall-lang/blob/f717c748b3751e5a1b22da8af3db31030e7dcd0d/standard/dhall.abnf#L348